presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -155,10 +158,10 @@ private void writePages(Iterator<Page> pageIterator)
         writable = false;
         try {
             InputStream input = new FileInputStream(targetFileName.toFile());
+            Closeable resources = closer.register(combineCloseables(input, () -> memoryContext.setBytes(0)));
```

Comment:
We need two closers here. Once that will be attached to the returned iterator, so that file input stream is closed upon iterator exhaustion. The `this.closer` is needed so that the file input stream is closed no matter what.

Still, i can inline `combineCloseables` if you wish.